tests/test_call.py

```diff
@@ -347,6 +347,101 @@ def test_no_thresholds(tmp_path):
 
     assert path.isdir(output_path) == False
 
+def test_thresholds_many(tmp_path):
+    # A test where many thresholds are used.
+    config = {}
+
+    clusters_path = get_path("data/clusters/thresholds_many.tsv")
+    pairwise_distances_path = get_path("data/pairwise_distances/thresholds_many.tsv")
+    output_path = path.join(tmp_path, "test_out")
+
+    config["dists"] = pairwise_distances_path
+    config["rclusters"] = clusters_path
+    config["outdir"] = output_path
+    config["force"] = False
+
+    config["thresholds"] = "26,24,22,20,18,16,14,12,10,8,6,4,2,0"
+    config["thresh_map"] = None
+
+    config["method"] = "single"
+
+    config["sample_col"] = "id"
+    config["address_col"] = "address"
+    config["delimiter"] = "."
+
+    config["batch_size"] = 100
+
+    call(config)
+
+    assert path.isdir(output_path)
+
+    # Clusters
+    clusters_path = path.join(output_path, "results.text")
+    assert path.isfile(clusters_path)
+    with open(clusters_path) as clusters_file:
+        clusters = csv.reader(clusters_file, delimiter="\t")
+
+        # The new f is the same as the existing a (1.1.1.1.1.1.1.1.1.1.1.1.1.1)
+        # The new g is the same as the existing e (1.1.1.3.3.4.4.4.4.4.4.4.4.4)
```

Comment:
Could you add another test case where there are new cluster addresses created instead of assigning samples to existing cluster addresses? That way we can make sure that when new cluster addresses are created, the property of numbers to the right never being lower than numbers to the left holds true.

Reply:
This was added after your comment, but this test from #42 adds a new address:

https://github.com/phac-nml/genomic_address_service/blob/1d390ea1395961ddeee5b79bdeb3be8178f243de/tests/test_call.py#L1679-L1755

Reply:
I've changed the distances so that `f` remains the same (same as `a`), but `g` now is completely different: 488a7b9

```
query_id	ref_id	dist
f	a	0
f	b	17
f	c	21
f	d	31
f	e	23
f	f	0
f	g	23
g	a	123
g	b	121
g	c	139
g	d	143
g	e	100
g	f	123
g	g	0
```

```python
assert ["id", "address"] in clusters
assert ["a", "1.1.1.1.1.1.1.1.1.1.1.1.1.1"] in clusters
assert ["b", "1.1.1.1.1.2.2.2.2.2.2.2.2.2"] in clusters
assert ["c", "1.1.1.2.2.3.3.3.3.3.3.3.3.3"] in clusters
assert ["d", "2.2.2.4.4.5.5.5.5.5.5.5.5.5"] in clusters
assert ["e", "1.1.1.3.3.4.4.4.4.4.4.4.4.4"] in clusters
assert ["f", "1.1.1.1.1.1.1.1.1.1.1.1.1.1"] in clusters
assert ["g", "3.3.3.5.5.6.6.6.6.6.6.6.6.6"] in clusters
```